### PR TITLE
Fix(preview middleware) preview client url

### PR DESCRIPTION
**Pull request by heimwege**
Using an absolute preview client url does not work in combination with [cds-plugin-ui5](https://www.npmjs.com/package/cds-plugin-ui5).

Sample:
- URL without cds-plugin-ui5: http://localhost:8080/localService/one/more/path/index.html#Shell-home (core resource url http://localhost:8080/resources/sap-ui-core.js)
- URL with cds-plugin-ui5: http://localhost:4004/com.sap.cap.fe.ts.sample/localService/one/more/path/index.html#Shell-home (core resource url http://localhost:4004/com.sap.cap.fe.ts.sample/resources/sap-ui-core.js)

This is fixed in this PR.

### CODE REVIEW

File: packages/preview-middleware/src/base/config.ts [hunk 1]
```diff
@@ -261,16 +262,17 @@ export function createFlpTemplateConfig(config: FlpConfig, manifest: Partial<Man
...
+        basePath: basePath,
         apps: {},
         init: config.init ? ns + config.init : undefined,
         ui5: {
             libs: getUI5Libs(manifest),
             theme: ui5Theme,
             flex,
             resources: {
-                [PREVIEW_URL.client.ns]: PREVIEW_URL.client.url
+                [PREVIEW_URL.client.ns]: PREVIEW_URL.client.getUrl(basePath)
```
Comment (heimwege) [inline packages/preview-middleware/src/base/config.ts:275]: An alternative would be to keep PREVIEW_URL as is and add the basePath here
`[PREVIEW_URL.client.ns]: posix.join(basePath, PREVIEW_URL.client.url)`
Not sure which one is better 🤷🏻

**Comment by heimwege** [comment]
Now I'm being blamed from the Inclusive Language Checker for the main branch name of OpenUI5. That's not fair 😿

File: packages/preview-middleware/src/base/config.ts (hunk 1 above)
Comment (tobiasqueck) [reply packages/preview-middleware/src/base/config.ts:275]: It's a matter of taste, go for what you prefer.

Comment (heimwege) [reply packages/preview-middleware/src/base/config.ts:275]: OK then let's leave it as it currently is. This hopefully reduces the risk that someone else just uses the path als URL without basePath

**Comment by tobiasqueck** [comment]
@heimwege while you were fixing the occurrences of the absolute url, someone added another one in `main` at https://github.com/SAP/open-ux-tools/blob/main/packages/preview-middleware/templates/flp/sandbox.html#L43. Could you please also handle it?

**Comment by heimwege** [comment]
> @heimwege while you were fixing the occurrences of the absolute url, someone added another one in `main` at https://github.com/SAP/open-ux-tools/blob/main/packages/preview-middleware/templates/flp/sandbox.html#L43. Could you please also handle it?

I adjusted that one but still it does not run with cds-plugin-ui because there seem to be other locations as well: [bootstrap.js](https://github.com/SAP/open-ux-tools/blob/main/packages/preview-middleware-client/src/flp/bootstrap.js#L12) 🙈

File: packages/preview-middleware-client/src/flp/bootstrap.js [hunk 2]
```diff
@@ -9,14 +9,18 @@
  * @param fnCallback {Function} The callback function to be executed after the bootstrap is loaded.
  */
 async function ushellBootstrap(fnCallback) {
-    let src = '/test-resources/sap/ushell/bootstrap/sandbox.js';
+    const currentPath = location.pathname.substring(0, location.pathname.lastIndexOf('/'));
```
Comment (heimwege) [inline packages/preview-middleware-client/src/flp/bootstrap.js:12]: Not sure if that's the best way to do this but at least it seems to work with cds-plugin-ui5 as well as without it (unit tests are still green 🐱). Basic idea taken from [testsuite.qunit.js template](https://github.com/SAP/open-ux-tools/blob/main/packages/preview-middleware/templates/test/testsuite.qunit.js#L3)

Comment (tobiasqueck) [reply packages/preview-middleware-client/src/flp/bootstrap.js:12]: Not beautiful but I don't have a smart suggestion, so I am ok with it.

**Review by tobiasqueck — APPROVED** [review]
* changeset ok
* code adheres to coding standards
* important additional info in the readme
* didn't locally test but checked the changes of the snapshots and they meet expectations

**Review by tobiasqueck — APPROVED** [review]
reviewed changes since last approval: all as expected

**Review by voicis — APPROVED** [review]
- changeset present
- code adheres to the coding standards
- good test coverage
- didn't test locally

**Pull request merged**